### PR TITLE
local package with build dependencies

### LINKED ISSUE #2531: Correct prod and dev dependencies in package.json
**Is your feature request related to a problem? Please describe.**
We are listing all deps we need to build the application in production deps and those we use only for dev in dev deps. This isn't quite right.

**Describe the solution you'd like**
The only dependencies that should be in the main dependency list are ones used at runtime. Therefore the dev dependencies aren't only ones used in development but also ones used during build stage.

**Describe alternatives you've considered**
n/a

**Testing notes**
Testing of all parts needed, probably worth a deploy test by pushing the branch to the test environment.

**Additional context**
Suggest the implementor checks best practices re non-obvious examples. For example, babel is a dev dependency but some of it's libraries may be used at runtime.

Also NB this almost certainly also requires our TravisCI and Jenkins files being updated.

### PR DESCRIPTION
**Pull request by thekp**
Resolves #2531 

**Overall change:** Use a local node package (/builderHelper) that contains our build dependencies so that we do not have to keep them in our production dependencies. 

`npm explore <package> -- <script from pacakge>` allows us to run scripts from the specified package.

**Code changes:**

- _created a local package that contains all our build dependencies so that we can separate it from our production dependencies._
- _create a build:helper script that runs `npm explore` to npm install the required build dependencies in the builderHelper package._

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
     "audit:ci": "audit-ci --low",
     "bbcA11y": "wait-on -t 20000 http://localhost.bbc.com:7080/news/articles/c6v11qzyv8po && bbc-a11y",
     "build:ci": "export NODE_ENV=production && rm -rf build && npm run build:test && npm run build:live",
-    "build": "rm -rf build && cp envConfig/local.env .env && npm run webpack:prod && npm run test:size",
+    "build:dependency": "npm explore 'packages/buildDependencies'",
```
Comment (jamesbhobbs) [inline package.json]: I think this should be prebuild, also in the explore script you need to refer to a script inside that packages package.json

File: package.json [hunk 2]
```diff
@@ -8,7 +8,8 @@
...
+    "build": "npm run build:dependency && rm -rf build && cp envConfig/local.env .env && npm run webpack:prod && npm run test:size",
```
Comment (jamesbhobbs) [inline package.json]: By making the other prebuild you should not need to change this

File: package.json [hunk 3]
```diff
@@ -111,23 +110,16 @@
     "@bbc/psammead-styles": "^1.1.5",
     "@bbc/psammead-timestamp-container": "^2.1.0",
     "@bbc/psammead-visually-hidden-text": "^1.1.0",
-    "assets-webpack-plugin": "^3.9.8",
     "babel-loader": "^8.0.5",
-    "babel-plugin-dynamic-import-node": "^2.3.0",
-    "babel-plugin-styled-components": "^1.10.6",
-    "babel-plugin-transform-react-remove-prop-types": "^0.4.24",
-    "brotli-webpack-plugin": "^1.0.0",
+    "build-helper": "file:buildHelper",
```
Comment (jamesbhobbs) [inline package.json]: Should be in dev deps

File: package.json [hunk 4]
```diff
@@ -8,7 +8,8 @@
     "audit:ci": "audit-ci --low",
     "bbcA11y": "wait-on -t 20000 http://localhost.bbc.com:7080/news/articles/c6v11qzyv8po && bbc-a11y",
     "build:ci": "export NODE_ENV=production && rm -rf build && npm run build:test && npm run build:live",
-    "build": "rm -rf build && cp envConfig/local.env .env && npm run webpack:prod && npm run test:size",
+    "build:helper": "npm isntall ./buildHelper && npm explore build-helper -- npm install",
```
Comment (jamesbhobbs) [inline package.json]: typo on install

File: package.json [hunk 5]
```diff
@@ -165,12 +151,16 @@
     "@testing-library/react": "^8.0.1",
     "@testing-library/react-hooks": "^1.1.0",
     "amphtml-validator": "^1.0.23",
+    "assets-webpack-plugin": "^3.9.10",
```
Comment (jamesbhobbs) [inline package.json]: all these dev deps except prop-types should be in the build package

File: package.json [hunk 6]
```diff
@@ -169,6 +154,7 @@
     "babel-eslint": "^10.0.2",
     "babel-jest": "^24.8.0",
     "bbc-a11y": "^2.3.0",
+    "build-helper": "file:buildHelper",
```
Comment (jamesbhobbs) [inline package.json:157]:
```suggestion
    "build-helper": "file:./buildHelper",
```

**Comment by thekp** [comment]
so my PR install the webpack dependencies in the new local package and not the root package but we run webpack from within our root, so it’s not using the modules from my new package.

I have tried updating the `requires` in our webpack config to use the modules from `build-helper` but then i get the web-cli not installed issue. and if i do install web-ci then it won’t find the other webpack dependencies becasue they are in `build-helper`

**Pull request closed without merging**